src/CommunityToolkit/Xamarin.CommunityToolkit.UnitTests/Converters/BaseConverterOneWay_Tests.cs

```diff
@@ -0,0 +1,97 @@
+{
+	[TestOf(typeof(BaseConverterOneWay<string, Color>))]
+	public class BaseConverterOneWay_Tests
+	{
+		#region AllowsNullOrDefault = true
+
+		/// <summary>
+		/// Derivation of <see cref="BaseConverterOneWay{TFrom,TTo}"/> to test the new <see cref="AllowsNullOrDefault"/> handling, allowing null.
+		/// </summary>
+		private class ColorNameToColorConverter : BaseConverterOneWay<string, Color>
+		{
+			protected override bool AllowsNullOrDefault => true;
+
+			public override Color ConvertFrom(string? value) =>
+				value switch
+				{
+					// By setting AllowsNullOrDefault to true, it is possible to handle this case by the derived class.
+					null => Color.Black,
+					"Red" => Color.Red,
+					"Blue" => Color.Blue,
+					_ => throw new ArgumentException($"{value} unknown.")
+				};
+		}
+
+		static IEnumerable<object?[]> GetTestDataWithNull()
+		{
+			yield return new object?[] { "Red", Color.Red };
+			yield return new object?[] { "Blue", Color.Blue };
+			yield return new object?[] { null, Color.Black };
+		}
+
+		[TestCaseSource(nameof(GetTestDataWithNull))]
+		public void ColorNameToColorConverterConvert(string value, Color expectedResult)
+		{
+			var colorNameToColorConverter = new ColorNameToColorConverter();
+			var result = colorNameToColorConverter.Convert(value, typeof(Color), null!, CultureInfo.CurrentCulture);
+
+			Assert.AreEqual(result, expectedResult);
+		}
+
+		[TestCase("")]
+		[TestCase("Green")]
+		[TestCase("red")]
+		public void ColorNameToColorConverterConvertInvalidValuesThrowArgumentException(string value)
+		{
+			var colorNameToColorConverter = new ColorNameToColorConverter();
+			Assert.Throws<ArgumentException>(() => colorNameToColorConverter.Convert(value, typeof(Color), null!, CultureInfo.CurrentCulture));
+		}
+
+		#endregion
+
+		#region AllowsNullOrDefault = false
+
+		/// <summary>
+		/// Derivation of <see cref="BaseConverterOneWay{TFrom,TTo}"/> to test the new <see cref="AllowsNullOrDefault"/> handling, disallowing null.
+		/// </summary>
+		private class ColorNameToColorConverterDisallowingNull : ColorNameToColorConverter
+		{
+			// By setting AllowsNullOrDefault to false, it is no longer possible to handle null and instead we expect an ArgumentException.
+			protected override bool AllowsNullOrDefault => false;
+		}
+
+		static IEnumerable<object?[]> GetTestDataWithoutNull()
+		{
+			yield return new object?[] { "Red", Color.Red };
+			yield return new object?[] { "Blue", Color.Blue };
+		}
+
+		[TestCaseSource(nameof(GetTestDataWithoutNull))]
+		public void ColorNameToColorConverterDisallowingNullConvert(string value, Color expectedResult)
+		{
+			var colorNameToColorConverterDisallowingNull = new ColorNameToColorConverterDisallowingNull();
+			var result = colorNameToColorConverterDisallowingNull.Convert(value, typeof(Color), null!, CultureInfo.CurrentCulture);
+
+			Assert.AreEqual(result, expectedResult);
+		}
+
+		[TestCase("")]
+		[TestCase("Green")]
+		[TestCase("red")]
+		[TestCase(null)]
```

Comment:
I think this could use your `GetTestDataWithNull` data source?

I am slightly confused by this test and I must be missing something, I read it that `"red"` and `"Green"` should throw an `ArgumentException` but these are valid values. Am I having a meltdown and reading this incorrectly 😃 ?

Reply:
"Green" is not defined in the converter.
"red" is lower case, which is only defined as "Red" (Pascal Case) in the converter (the tests are meaningless because we don't need to test the actual converter's logic but wanted them to look like a real test would look like).

If you want I can make it more logical and test for any `Forms.Color` (OrdinalIgnoreCase), but that's a bit more complex, I think.

This can't use `GetTestDataWithNull`, because these are "invalid argument" tests. I can't use Color.* (as expectedResult) in `TestCase`s otherwise I would not have used `TestCaseSource` for the valid tests and just used `TestCase`s.

I could make it more general and return `(string value, Color expectedValue, bool expectsError)` and just use one test (testing both valid and invalid arguments), I've copied most of this from the other converter tests to stay consistent.

Reply:
Thanks for explaining it, I think this just shows I probably need a break from my screen haha.

I am fine with keeping this mock converter as simple as possible so ignore my point about the data source.